### PR TITLE
fix(ci): Fix for issue 9474 to Avoid posting test results if tests are skipped

### LINKED ISSUE #9474: Avoid posting test results if tests are skipped
On this PR(https://github.com/magma/magma/pull/9468),  the agw test for example has been skipped but the bot still post the test result as a comment.
We should not post test results if there are none.

### PR DESCRIPTION
**Pull request by ranj8work**
## Summary
Fix for issue 9474 to Avoid posting test results if tests are skipped

## Test Plan
Tested in my forked repo under branch test-9474-testresults. 
Successful Unit test results workflow execution is [here](https://github.com/ranj8work/magma/actions/runs/1388267663).
Tested with changes for nms workflow is [here](https://github.com/ranj8work/magma/actions/runs/1388223263).

## Additional Information

- [NO] This change is backwards-breaking

### CODE REVIEW

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
Instead of storing not skipped  as a variable, could you store it as skipped.
Don't like to have 'negative' variable.

File: .github/workflows/python-workflow.yml [hunk 1]
```diff
@@ -22,6 +22,18 @@ jobs:
           filters: |
             filesChanged:
               - [".github/workflows/python-workflow.yml", "**/*.py"]
+      # Need to save PR number as Github action does not propagate it with workflow_run event
+      - name: Save PR number
+        if: always()
+        run: |
+          mkdir -p ./pr
+          echo -n ${{ github.event.number }} > ./pr/NR
+          echo -n "false" > ./pr/RPR
```
Comment (quentinDERORY) [inline .github/workflows/python-workflow.yml:42]: looks like you missed to export the skipped variable ?

File: .github/workflows/cloud-workflow.yml [hunk 2]
```diff
@@ -25,6 +25,16 @@ jobs:
           filters: |
             filesChanged:
               - [".github/workflows/cloud-workflow.yml", "cwf/cloud/**", "fbinternal/cloud/**", "feg/cloud/**", "lte/cloud/**", "orc8r/cloud/**", "orc8r/lib/**"]
+      - name: Save should_not_skip output
+        if: always()
+        run: |
+          mkdir -p ./pr
+          echo -n ${{ !steps.changes.outputs.filesChanged }} > ./pr/skipped
```
Comment (quentinDERORY) [inline .github/workflows/cloud-workflow.yml]: replace !steps.changes.outputs.filesChanged with  steps.changes.outputs.filesChanged == 'false'

File: .github/workflows/golang-build-test.yml [hunk 3]
```diff
@@ -22,6 +22,16 @@ jobs:
           filters: |
             filesChanged:
               - [".github/workflows/golang-build-test.yml", "src/go/**"]
+      - name: Save should_not_skip output
+        if: always()
+        run: |
+          mkdir -p ./pr
+          echo -n ${{ !steps.changes.outputs.filesChanged }} > ./pr/skipped
```
Comment (quentinDERORY) [inline .github/workflows/golang-build-test.yml]: here too

File: .github/workflows/nms-workflow.yml [hunk 4]
```diff
@@ -24,6 +24,17 @@ jobs:
           filters: |
             filesChanged:
               - [".github/workflows/nms-workflow.yml", "nms/**"]
+      - name: Save should_not_skip output
+        if: always()
+        run: |
+          mkdir -p ./pr
+          echo -n ${{ !steps.changes.outputs.filesChanged }} > ./pr/skipped
```
Comment (quentinDERORY) [inline .github/workflows/nms-workflow.yml]: here too

File: .github/workflows/python-workflow.yml (hunk 1 above)
Comment (ranj8work) [reply .github/workflows/python-workflow.yml:42]: No @quentinDERORY 
I had to keep place holder for the RPR (Reverted PR) so it wont fail in comment-pr-on-check-failure.yml workflow which checks for RPR Reverted PR) for DCO checks and Semantic PR checks. Do I make sense.

File: .github/workflows/nms-workflow.yml (hunk 4 above)
Comment (ranj8work) [reply .github/workflows/nms-workflow.yml]: @quentinDERORY made changes are rebased

File: .github/workflows/golang-build-test.yml (hunk 3 above)
Comment (ranj8work) [reply .github/workflows/golang-build-test.yml]: @quentinDERORY made changes and rebased.

**Comment by electronjoe** [comment]
Not sure what's going on here but presume that a git rebase has gone awry - this massive delta PR has added the ~world as reviewers (see 1_400 files modified, 30k LOC). You might begin by moving to draft mode on the PR to clarify for reviewers that the PR is not ready for review (given it's current state).

**Pull request closed without merging**